### PR TITLE
Small cleanups to make lint and TS compiler happier

### PR DESCRIPTION
**Pull request by jtakalai**
Changes mostly in data union code, in how ethers is imported

### CODE REVIEW

File: packages/client/tsconfig.json [hunk 1]
```diff
@@ -2,7 +2,8 @@
     "extends": "@streamr/dev-config/ts/tsconfig.browser.json",
     "compilerOptions": {
         "outDir": "dist",
-        "rootDirs": ["src", "test"]
+        "rootDirs": ["src", "test"],
+        "resolveJsonModule": true
```
Comment (timoxley) [inline packages/client/tsconfig.json]: `resolveJsonModule` shouldn't be necessary, did you hit an issue? https://github.com/streamr-dev/network-monorepo/blob/193e6eac1e150cd7b3f1a20a7c86e814bcb12dc0/packages/dev-config/ts/tsconfig.browser.json#L22

**Review by timoxley — APPROVED** [review]
Great stuff

File: packages/client/tsconfig.json (hunk 1 above)
Comment (kare) [reply packages/client/tsconfig.json]: It's set in @streamr/dev-config: https://github.com/streamr-dev/network-monorepo/blob/main/packages/dev-config/ts/tsconfig.browser.json#L22

File: packages/client/tsconfig.json [hunk 2]
```diff
@@ -2,7 +2,8 @@
     "extends": "@streamr/dev-config/ts/tsconfig.browser.json",
     "compilerOptions": {
         "outDir": "dist",
-        "rootDirs": ["src", "test"]
+        "rootDirs": ["src", "test"],
```
Comment (kare) [inline packages/client/tsconfig.json]: @jtakalai @timoxley @harbu @teogeb
Any reason for not using these configuration settings?
```
    "incremental": true,                              /* Enable incremental compilation */
    "composite": true,                                /* Enable constraints that allow a TypeScript project to be used with project references. */
    "tsBuildInfoFile": ".tsbuildinfo",                /* Specify the folder for .tsbuildinfo incremental compilation files. */
```
Is the `.tsbuildinfo` cacheable in CI?

Comment (kare) [reply packages/client/tsconfig.json]: I see `incremental` is set in the config, but it requires `composite = true`

File: packages/client/tsconfig.json (hunk 1 above)
Comment (jtakalai) [reply packages/client/tsconfig.json]: I think tests import contract jsons directly

Comment (jtakalai) [reply packages/client/tsconfig.json]: I can delete it, too

File: packages/client/tsconfig.json (hunk 2 above)
Comment (jtakalai) [reply packages/client/tsconfig.json]: no idea

**Pull request merged**